hazelcast/src/main/java/com/hazelcast/instance/IHazelcastInstanceFactory.java

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright (c) 2008-2015, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.instance;
+
+import java.util.Collection;
+import java.util.Map;
+import java.util.Set;
+
+import com.hazelcast.config.Config;
+import com.hazelcast.core.HazelcastInstance;
+import com.hazelcast.core.Member;
+
+public interface IHazelcastInstanceFactory {
+    Set<HazelcastInstance> getAllHazelcastInstances();
+
+    HazelcastInstance getHazelcastInstance(String instanceName);
+
+    HazelcastInstance getOrCreateHazelcastInstance(Config config);
+
+    HazelcastInstance newHazelcastInstance(Config config);
+
+    HazelcastInstance newHazelcastInstance(Config config, String instanceName,
+                                           NodeContext nodeContext);
+
+    Set<HazelcastInstanceImpl> getInstanceImpls(Collection<Member> members);
+
+    void remove(HazelcastInstanceImpl instance);
+
+    void shutdownAll();
+
+    void terminateAll();
+
+    Map<MemberImpl, HazelcastInstanceImpl> getInstanceImplMap();
```

Comment:
I would move this method underneath the 'getInstanceImp' so that is closely placed to another related method 'getInstanceImpls'